### PR TITLE
[SPARK-19809][SQL][TEST][FOLLOWUP] Move the test case to HiveOrcQuerySuite

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

As a follow-up of #19948 , this PR moves the test case and adds comments.

## How was this patch tested?

Pass the Jenkins.

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -2172,21 +2172,4 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       }
     }
   }
-
-  test("SPARK-19809 NullPointerException on zero-size ORC file") {
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Should we move the test above `SPARK-22267 Spark SQL incorrectly reads ORC files when column order is different` too? I think basically that's also the same instance.

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Yep. I'll piggy back here.

**Review by HyukjinKwon — APPROVED** [review]
cc @gatorsmile

**Comment by dongjoon-hyun** [comment]
Thank you always for review and approval, @HyukjinKwon !

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @gatorsmile .

**Pull request closed without merging**